libsolidity/analysis/OverrideChecker.cpp

```diff
@@ -136,37 +380,20 @@ void OverrideChecker::check(ContractDefinition const& _contract)
 
 void OverrideChecker::checkIllegalOverrides(ContractDefinition const& _contract)
 {
-	FunctionMultiSet const& inheritedFuncs = inheritedFunctions(_contract);
-	ModifierMultiSet const& inheritedMods = inheritedModifiers(_contract);
+	OverrideProxyBySignatureMultiSet const& inheritedFuncs = inheritedFunctions(_contract);
+	OverrideProxyBySignatureMultiSet const& inheritedMods = inheritedModifiers(_contract);
+	OverrideProxyBySignatureMultiSet const& inheritedVars = inheritedPublicStateVariables(_contract);
+	OverrideProxyBySignatureMultiSet inheritedFuncsAndVars;
+	// TODO maybe better to include public state variables right away in inheritedFunction()?
```

Comment:
I think it's fine to keep it separated.

Reply:
I merged them since they were manually merged for all use-cases.